src/game/start.py

```diff
@@ -1,99 +1,135 @@
-    try:
-        parameter = json.loads(parameter)
-    except json.JSONDecodeError:
-        parameter = ""
-        error_message = "[The state to load is not wellformed] : Ignored"
-
-    return parameter, error_message
-
-
-def grid_to_string(grid):
-    display_string = '    A   B   C   D\n'
-    i = 1
-    for line in grid:
-        display_string += ' '
-        display_string += str(i)
-        display_string += ' '
-        for position in line:
-            position = str(position)
-            if len(position) < 2:
-                display_string += ' '
-            display_string += position
-            display_string += '  '
-        display_string += '\n'
-        i += 1
-    return display_string
-
-
-def pieces_to_string(remaining_pieces, game_turn):
-    display_string = 'Remaining pieces :\n'
-    for i in range(1, PIECES_NUMBER + 1):
-        display_string += ' '
-        if remaining_pieces.count(i):
-            display_string += selected_piece_to_string(i, game_turn)
-        else:
-            display_string += ' . '
-        display_string += ' '
-    return display_string
-
-
-def selected_piece_to_string(piece_number, game_turn):
-    if game_turn.selected_piece == piece_number:
-        return "[" + str(piece_number) + "]"
-    return str(piece_number)
-
-
-def players_to_string(game_turn):
-    player_1 = selected_player_to_string('Player 1', game_turn.player_one_active)
-    player_2 = selected_player_to_string('Player 2', not game_turn.player_one_active)
-    return player_1 + "     " + player_2
-
-
-def selected_player_to_string(player_name, selected):
-    if selected:
-        player_name = "=> " + player_name + " <="
-    return player_name
-
-
-def display_game(game_state):
-    print()
-    print(grid_to_string(game_state.grid))
-    print()
-    print(players_to_string(game_state.game_turn))
-    print()
-    print(pieces_to_string(game_state.remaining_pieces, game_state.game_turn))
+class UIRender:
+
+    """Definition of the user interface and the interactions:"""
+
+    def start_game(self):
+        print("Welcome to Quarto-Py")
+
+        initial_state, error_parsing = self.get_state_parameter(sys.argv)
+        pieces_list = Piece.create_pieces_list()
+
+        game_state = GameState()
+
+        if len(error_parsing) > 0:
+            game_state.message = error_parsing
+        elif type(initial_state) is dict:
+            game_state.load_state(initial_state)
+
+        if (len(game_state.message) > 0):
+            game_state.message += """\nValid sample : --state='{"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 1,"selected" : 7}}'"""
+
+        while len(game_state.remaining_pieces) > 0:
+            self.display_game(game_state)
+            self.prompt_piece_selection(game_state)
+            game_state.swich_player()
+            self.display_game(game_state)
+            self.prompt_piece_location(game_state)
+
+    def prompt_piece_selection(self, game_state):
```

Comment:
Much better function ! It's readable and understandable. Good work :1st_place_medal: